src/StrawberryfieldFilePersisterService.php

```diff
@@ -223,7 +231,10 @@ public function getDestinationUri(
   ) {
 
     // Default $relativefolder is a 3 char hash generated by a checksum algorithm.
-    $relativefolder = substr($checksum, 0, 3);
+    // First, get any parent directories.
+    $relativefolders = explode('/', $this->storageConfig->get('file_path') ?? "");
+    $relativefolders[] = substr($checksum, 0, 3);
+    $relativefolder = implode('/', array_filter($relativefolders));
```

Comment:
Ok, this needs to be added to the refactor too

Reply:
I'm thinking let's get your refactor merged first, and then resolve this branch it. Does that make sense?

Reply:
Ok, let's roll with that. Will wait for your review on that pull before merging it. Thx